tests/S3/S3Transfer/S3TransferManagerTest.php

```diff
@@ -0,0 +1,3869 @@
+        $targetDirectory = $baseDirectory . DIRECTORY_SEPARATOR . $destination;
+        if (is_dir($targetDirectory)) {
+            TestsUtility::cleanUpDir($targetDirectory);
+        }
+
+        mkdir($targetDirectory, 0777, true);
+
+        // Get prefix so it can be used in body creation for ListObjectsV2
+        $prefix = $config['prefix'] ?? '';
+        // Prepare the pool of responses
+        $s3Client = $this->getS3ClientWithSequentialResponses(
+            array_map(function ($expectation) {
+                $operation = $expectation['request']['operation'];
+
+                return array_merge(
+                    $expectation['response'],
+                    ['operation' => $operation]
+                );
+            }, $expectations),
+            function (
+                string $operation,
+                array|string|null $body,
+                ?array &$headers
+            ) use ($prefix, $bucket): StreamInterface
+            {
+                if ($operation === 'ListObjectsV2') {
+                    $listObjectsV2Template = self::$s3BodyTemplates[$operation];
+                    $listObjectsV2ContentsTemplate = self::$s3BodyTemplates[
+                        $operation . "::Contents"
+                    ];
+                    $bodyBuilder = str_replace(
+                        "{{Bucket}}",
+                        $bucket,
+                        $listObjectsV2Template
+                    );
+                    $bodyBuilder = str_replace(
+                        "{{Prefix}}",
+                        $prefix,
+                        $bodyBuilder
+                    );
+
+
+                    $itemBuilder = "";
+                    foreach ($body as $item) {
+                        $itemBuilder = $itemBuilder . "\n$listObjectsV2ContentsTemplate";
+                        $itemBuilder = str_replace(
+                            ['{Key}', '{Size}'],
+                            [$item['key'], $item['size']],
+                            $itemBuilder
+                        );
+                    }
+
+                    $bodyBuilder = str_replace(
+                        ['{Contents}'],
+                        [$itemBuilder],
+                        $bodyBuilder
+                    );
+
+                    $fixedBody = Utils::streamFor($bodyBuilder);
+                    $body = null;
+                } else {
+                    $fixedBody = Utils::streamFor(
+                        str_repeat(
+                            '*',
+                            $headers['Content-Length']
+                        )
+                    );
+                    $body = null;
+                }
+
+                if (isset($headers['ChecksumAlgorithm'])) {
+                    // Checksum injection when expected to succeed at checksum validation
+                    // This is needed because the checksum in the test is wrong
```

Comment:
Which checksums are wrong?

Reply:
After reviewing this test carefully I found that, in this scenario is ok to conditionally add a checksum, because the test is not trying to validate the checksum itself, but instead make sure that when a checksum is present a validation is done. Also, the test in the fixtures does not provide a body, which is also mocked by the implementation.